site/src/sphinx/advanced-saml.rst

```diff
@@ -1,6 +1,187 @@
+.. _`OpenSAML`: https://wiki.shibboleth.net/confluence/display/OS30/Home
+.. _`Security Assertion Markup Language (SAML)`: https://en.wikipedia.org/wiki/Security_Assertion_Markup_Language
+
 .. _advanced-saml:
 
 SAML Single Sign-On
 ===================
 
-TBW - See :api:`SamlServiceProvider` and :api:`SamlServiceProviderBuilder`
+.. note::
+
+    Visit `armeria-examples <https://github.com/line/armeria-examples>`_ to find the fully working example code.
+
+What is SAML?
+-------------
+
+`Security Assertion Markup Language (SAML)`_ is an open standard for exchanging authentication and authorization
+data between an identity provider and a service provider. In this protocol, a service provider is an endpoint
+which provides a web service to an end user, and an identity provider is in charge of authenticating an end
+user with information sent by the service provider.
+Armeria currently provides OpenSAML_ based ``armeria-saml`` module in order to support the integration with
+an identity provider from a service provider's point of view.
+
+Configuring your server as a service provider
+---------------------------------------------
+
+The first step to configure a service provider is adding ``armeria-saml`` to your dependencies.
+
+For Maven:
+
+.. parsed-literal::
+    :class: highlight-xml
+
+    <dependency>
+        <groupId>com.linecorp.armeria</groupId>
+        <artifactId>saml</artifactId>
+        <version>\ |release|\ </version>
+    </dependency>
+
+For Gradle:
+
+.. parsed-literal::
+    :class: highlight-groovy
+
+    dependencies {
+        compile 'com.linecorp.armeria:saml:\ |release|\ '
+    }
+
+After that, you need to prepare your keystore file which contains a key pair for signing and encryption
+of a SAML message. Also, you need to import the certificate of your identity provider into the keystore
+which contains your key pairs. In this example, we are using an identity provider service hosted by
+`SSO Circle <https://www.ssocircle.com/en/>`_ in order to authenticate an end user. The following commands
+may help you to get a keystore.
+
+.. code-block:: bash
+
+    # Generate a new key pair as alias 'signing'.
+    keytool -genkeypair -keystore sample.jks -storepass 'N5^X[hvG' -keyalg rsa -sigalg sha1withrsa
+      -dname 'CN=Unknown, OU=Unknown, O=Unknown, L=Unknown, ST=Unknown, C=Unknown' -alias signing
+
+    # Import a certificate into the keystore as alias 'https://idp.ssocircle.com', which is the entity ID
+    # of the SSO Circle.
+    keytool -importcert -keystore sample.jks -storepass 'N5^X[hvG' -file ssocircle.cert
```

Comment:
Where do a user download `ssocircle.cert`? How about placing `wget`, `curl` or `openssl` command that downloads the certificate file before this line?

Reply:
Mentioned a URL that you found. Thanks!